docs/content/tutorial/step_03.ngdoc

```diff
@@ -1,223 +1,279 @@
+DOM sub-tree constructed using the provided `template` and managed by an instance of the specified
+controller.
 
-In Step 2, we learned how to write and run unit tests. Unit tests are perfect for testing
-controllers and other components of our application written in JavaScript, but they can't easily
-test DOM manipulation or the wiring of our application. For these, an end-to-end test is a much
-better choice.
+But wait, where did that `$ctrl` come from and what does it refer to ?
 
-The search feature was fully implemented via templates and data-binding, so we'll write our first
-end-to-end test, to verify that the feature works.
+For reasons already mentioned (and for other reasons that are out of the scope of this tutorial), it
+is considered a good practice to avoid using the scope directly. We can (and should) use our
+controller instance; i.e. assign our data and methods on properties of our controller (the "`this`"
+inside the controller constructor), instead of directly to the scope.
 
-__`test/e2e/scenarios.js`:__
+From the template, we can refer to our controller instance using an alias. This way, the context of
+evaluation for our expressions is even more clear. By default, components use `$ctrl` as the
+controller alias, but we can override it, should the need arise. (It is a very uncommon need though,
+so we will be sticking with `$ctrl` most of the time.)
 
-```js
-describe('PhoneCat App', function() {
+There are more options available, so make sure you check out the
+{@link ng.$compileProvider#component API Reference}, before using `.component()` in your own
+applications.
 
-  describe('Phone list view', function() {
 
-    beforeEach(function() {
-      browser.get('app/index.html');
-    });
+## Using Components
 
+Now that we know how to create components, let's refactor the HTML page to make use of our newly
+acquired skill.
 
-    it('should filter the phone list as a user types into the search box', function() {
+<br />
+**`app/index.html`:**
 
-      var phoneList = element.all(by.repeater('phone in phones'));
-      var query = element(by.model('query'));
+```html
+<html ng-app="phonecatApp">
+<head>
+  ...
+  <script src="bower_components/angular/angular.js"></script>
+  <script src="app.js"></script>
+  <script src="phone-list.component.js"></script>
+</head>
+<body>
+
+  <!-- Use a custom component to render a list of phones -->
+  <phone-list></phone-list>
+
+</body>
+</html>
+```
 
-      expect(phoneList.count()).toBe(3);
+<br />
+**`app/app.js`:**
 
-      query.sendKeys('nexus');
-      expect(phoneList.count()).toBe(1);
+```js
+// Define the `phonecatApp` module
+angular.module('phonecatApp', []);
+```
 
-      query.clear();
-      query.sendKeys('motorola');
-      expect(phoneList.count()).toBe(2);
-    });
+<br />
+**`app/phone-list.component.js`:**
+
+```js
+// Register `phoneList` component, along with its associated controller and template
+angular.
+  module('phonecatApp').
```

Comment:
I don't think it is explicitly mentioned anywhere, but this seems to be what john's styleguide is going throughout.
This is how I do it too fwiw :stuck_out_tongue:

Pros and cons of one module per file:

> you don't have to sort the files when building a distribution file

:+1: 

> you can easily move a file to another project without having to bring along its module definition

The idea is that you are creating a module for the "least viable feature" - i.e. the smaller chunk of functionality that does something worth moving around or re-using in another project. So, there is no point in copying just the controller file of a component over to another project - you would copy the whole component directory instead(including the controller, controller spec, the module definition, the template etc).

> Too many modules

:-1:

> I don't think it's promoted as a good practice anywhere (at least I haven't seen it), so there might be few projects following this convention out there (I'm just speculating here).

:-1: 

--
Could go either way (but only because I'm too tempted by not having to sort the files when bundling/testing :stuck_out_tongue: )
I also have a slight concern about the overhead of creating a module for everything. I think there is a (tiny) bit of work involved in loading a module, which could add up.

Leaving it up to you :grin: